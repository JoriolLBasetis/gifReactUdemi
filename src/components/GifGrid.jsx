import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

    const {gifs,isLoading} = useFetchGifs(category);
    
    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && (<h2 >is loading</h2>)
            }
            
            <div className="card-grid">
                {gifs.map((gif) => (
                    <GifGridItem key={gif.id} {...gif} />
                ))
                }
            </div>
        </>
    )
}
