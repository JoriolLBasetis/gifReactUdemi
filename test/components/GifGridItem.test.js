import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components"

describe('GridItem tests', () => {
    const testUrl = 'https://test.com/gim.gif';
    const testTitle = 'title';

    test('snapshot match', () => {
        const { container } = render(<GifGridItem url={testUrl} title={testTitle} />);
        expect(container).toMatchSnapshot();
    })
    test('mostra imatge amb la url',()=>{
        render(<GifGridItem url={testUrl} title={testTitle} />);
        const {src, alt}= screen.getByRole('img');
        expect(src).toBe(testUrl);
        expect(alt).toBe(testTitle);
    })
    test('show title in component', ()=>{
        render(<GifGridItem url={testUrl} title={testTitle} />);
        expect(screen.getByRole('paragraph').innerHTML).toBe(testTitle);
    })
})