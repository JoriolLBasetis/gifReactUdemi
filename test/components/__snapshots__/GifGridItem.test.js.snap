// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem tests snapshot match 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://test.com/gim.gif"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
